src/Features/Core/Portable/RemoveUnnecessaryParentheses/AbstractRemoveUnnecessaryParenthesesDiagnosticAnalyzer.cs

```diff
@@ -126,8 +129,16 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
 
             var additionalLocations = ImmutableArray.Create(parenthesizedExpression.GetLocation());
 
-            // Fades the open parentheses character and reports the suggestion.
-            context.ReportDiagnostic(Diagnostic.Create(s_diagnosticWithFade, parenthesizedExpression.GetFirstToken().GetLocation(), additionalLocations));
+            // Since this diagnostic descriptor is not user configurable, do not report it if this diagnostic id is hidden / suppressed.
+            var effectiveSeverity = syntaxTree.GetEffectiveSeverity(IDEDiagnosticIds.RemoveUnnecessaryParenthesesDiagnosticId, context.SemanticModel.Compilation);
+            if (effectiveSeverity != ReportDiagnostic.Suppress && effectiveSeverity != ReportDiagnostic.Hidden)
```

Comment:
📝 Hidden diagnostics should still have fading

Reply:
Agreed - we should always report the fading diagnostic as long as `RemoveUnnecessaryParenthesesDiagnosticId` is not effectively suppressed.